Tasks/JavaToolInstaller/AzureStorageArtifacts/AzureStorageArtifactDownloader.ts

```diff
@@ -0,0 +1,89 @@
+import * as Q from 'q';
```

Comment:
Is this file copied wholesale from AzureStorageArtifactDownloader.ts in JenkinsDownloadArtifacts?  If so, you could potentially move this into common and have both tasks use the same file.

Reply:
I made some minor tweaks, but it could be probably be extracted out. I'll give that a try.